prometheus_juju_exporter/collector.py

```diff
@@ -153,14 +180,15 @@ async def _get_machine_stats(
             machine_type = self._get_machine_type(machine)
             value = self._get_gauge_value(status=machine["agent-status"]["status"])
 
+            hostname = self._get_machine_sensible_hostname(machine)
             labels = self._create_gauge_label(
-                hostname=machine["hostname"],
+                hostname=hostname,
                 model_name=model_name,
                 machine_type=machine_type.value,
             )
 
-            if labels["hostname"] and labels["hostname"] != "None":
-                self.currently_cached_labels[machine["hostname"]] = (
+            if labels["hostname"] != "None":
```

Comment:
Since `labels["hostname"]` and `hostname` are the same thing, maybe it would be cleaner to have `if hostname != "None"` here and move `_create_gauge_label()` inside the if condition

Reply:
yeah good idea thanks, will address in the update